src/androidMain/kotlin/com/liftric/job/queue/NetworkListener.kt

```diff
@@ -0,0 +1,41 @@
+package com.liftric.job.queue
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+actual class NetworkListener(
+    networkManager: NetworkManager,
+    scope: CoroutineScope = CoroutineScope(context = Dispatchers.Default)
+) : AbstractNetworkListener(
+    networkManager = networkManager,
+    scope = scope
+) {
+    private val _currentNetworkState = MutableStateFlow(NetworkState.NONE)
+    override val currentNetworkState: StateFlow<NetworkState>
+        get() = _currentNetworkState.asStateFlow()
+
+    override fun observeNetworkState() {
+        scope.launch {
+            networkManager
+                .observeNetworkConnection()
+                .collect { currentNetworkState ->
+                    _currentNetworkState.emit(
+                        when (currentNetworkState) {
+                            NetworkState.NONE -> NetworkState.NONE
+                            NetworkState.MOBILE -> NetworkState.MOBILE
+                            NetworkState.WIFI -> NetworkState.WIFI
+                            null -> NetworkState.NONE
+                        }
+                    )
+                }
+        }
+    }
+
+
+    // not needed for Android
+    override fun stopMonitoring() {}
```

Comment:
We could either implement that in a way that the state gets refreshed every 5 seconds (or in any other interval) to mimic the iOS behaviour a bit. But this is not subject to be implemented instantly but rather - if I understood everything right - to keep it as an issue for future improvements.

Reply:
Actually, I can just create a job variable and assign it in observeNetworkState. Then, I can simply cancel it. I will add that for now. I had a wrong thinking at first. I will also do the same for ios.

```suggestion
    override fun stopMonitoring() {
    job?.cancel()
    }
```